tidy3d/web/api/mode.py

```diff
@@ -83,6 +85,21 @@ def run(
     if verbose:
         console = get_logging_console()
 
+    if reduce_simulation == "auto":
+        sim = mode_solver.simulation
+        contains_custom = any(isinstance(s.medium, AbstractCustomMedium) for s in sim.structures)
+        contains_custom = contains_custom or isinstance(sim.medium, AbstractCustomMedium)
```

Comment:
We do something like that in a bunch of places but yeah it seems like there's still no in-built way to get "all media, including the simulation one" and "all structures, including the simulation as a structure"...

Reply:
Well, I guess I could've just used `Simulation.mediums`

Reply:
Ah yeah so there is something like that!